__init__.py

```diff
@@ -87,94 +218,68 @@ def handle_tell_more(self, message):
         if not summary:
             self.speak_dialog("thats all")
         else:
-            self.gui.clear()
-
-            try:
-                pagetext = wiki.page(results[0])
-            except wiki.PageError:
-                pagetext = wiki.page(results[0], auto_suggest=False)
-
-            self.gui['summary'] = summary
-            self.gui['imgLink'] = wiki_image(pagetext)
-            self.gui.show_page("WikipediaDelegate.qml", override_idle=60)
+            self.display_article(article)
             self.speak(summary)
-            self.set_context("wiki_article", article)
+            # Update context
+            self.set_context("wiki_article", article.serialize())
             self.set_context("spoken_lines", str(lines_spoken_already+5))
 
     @intent_handler("Random.intent")
     def handle_random_intent(self, message):
         """ Get a random wiki page.
 
-            Uses the Special:Random page of wikipedia
+        Uses the Special:Random page of wikipedia
         """
         # Talk to the user, as this can take a little time...
         search = wiki.random(pages=1)
         self.speak_dialog("searching", {"query": search})
-        self._lookup(search)
+        self.handle_result(wiki_lookup(search))
```

Comment:
Does this need a `lang_code` param.

Should we have a default value?

or should it actually call `self.get_wiki_result(search)`

Reply:
That is actually changed here locally to add the lang code, must have missed pushing the update, I think this will not need get_wiki_result as the random produces a valid page and the auto_suggest handling is not needed.